stylix/target.nix

```
@@ -34,28 +34,97 @@
+          isWrapped = builtins.match ''[(].*[)]'' trimmed != null;
+          hasNewlines = lib.strings.hasInfix "\n" trimmed;
+          needsWrapping = builtins.any (op: lib.strings.hasInfix op trimmed) [
+            # These operators have lower precedence than `&&`
+            # See https://nix.dev/manual/nix/2.28/language/operators
+            "||"
+            "->"
+            "|>"
+            "<|"
+            # These keywords would also need wrapping
+            "with "
+            "assert "
+          ];
+          indented =
+            if indentMultilineExpr then
+              lib.pipe trimmed [
+                (lib.strings.splitString "\n")
+                (map (line: if line == "" then "" else "  " + line))
+                (builtins.concatStringsSep "\n")
+              ]
+            else
+              trimmed;
+          wrapped = if hasNewlines then "(\n${indented}\n)" else "(${trimmed})";
+        in
+        if autoWrapExpr && !isWrapped && needsWrapping then wrapped else trimmed;
     in
     {
       mkEnableTarget =
-        humanName: autoEnable:
-        lib.mkEnableOption "theming for ${humanName}"
-        // {
-          default = cfg.autoEnable && autoEnable;
-          example = !autoEnable;
-        }
-        // lib.optionalAttrs autoEnable {
-          defaultText = lib.literalMD "same as `stylix.autoEnable`";
+        name: autoEnable:
+        config.lib.stylix.mkEnableTargetWith { inherit name autoEnable; };
+
+      mkEnableTargetWith =
+        {
+          name,
+          autoEnable ? true,
+          autoEnableExpr ? null,
+          autoWrapExpr ? true,
+          example ? if args ? autoEnableExpr then true else !autoEnable,
+        }@args:
+        let
+          wrapExpr = wrapExprWith {
+            inherit autoWrapExpr;
+          };
+        in
+        self.mkEnableIf {
+          description = "Whether to enable theming for ${name}";
+          condition = cfg.autoEnable && autoEnable;
+          conditionText =
+            if args ? autoEnableExpr then
+              lib.literalExpression "stylix.autoEnable && ${wrapExpr autoEnableExpr}"
+            else if autoEnable then
+              lib.literalExpression "stylix.autoEnable"
+            else
+              false;
+          inherit example;
         };
+
       mkEnableWallpaper =
         humanName: autoEnable:
-        lib.mkOption {
-          default = config.stylix.image != null && autoEnable;
-          example = config.stylix.image == null;
+        self.mkEnableIf {
           description = "Whether to set the wallpaper for ${humanName}.";
+          condition = config.stylix.image != null && autoEnable;
+          conditionText =
+            if autoEnable then lib.literalExpression "stylix.image != null" else false;
+          example = config.stylix.image == null;
+        };
+
+      mkEnableIf =
+        {
+          description,
+          condition,
+          conditionText ? null,
```

Would the following renames not make it more obvious what is being interfaced:

```suggestion
          default,
          defaultText ? null,
```

In that case, we might also want to rename the internal `autoEnable` argument to `default`.

---

> we might also want to rename the internal `autoEnable` argument to `default`

`autoEnable` is distinct from a default because it is only used when `config.stylix.autoEnable` is true.

Effectively, `autoEnable` is specifying whether or not to allow `config.stylix.autoEnable` to define the default value.[^1]

[^1]: Although for performance reasons, the `&&` is the other way around, logically speaking `&&` is symmetrical so this doesn't matter.